src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -334,6 +341,42 @@ private TypeSymbolWithAnnotations(TypeSymbol defaultType, NullableAnnotation nul
             _extensions = extensions;
         }
 
+        public TypeSymbolWithAnnotations AsSpeakable()
+        {
+            return Create(this.TypeSymbol, this.GetSpeakableNullableAnnotation(), this.CustomModifiers);
+        }
+
+        /// <summary>
+        /// This method projects nullable annotations onto a smaller set that can be expressed in source.
+        /// </summary>
+        public NullableAnnotation GetSpeakableNullableAnnotation()
+        {
+            var annotation = NullableAnnotation;
+            switch (annotation)
+            {
+                case NullableAnnotation.Unknown:
+                case NullableAnnotation.NotAnnotated:
+                case NullableAnnotation.Annotated:
+                    return annotation;
+
+                case NullableAnnotation.Nullable:
+                    if (TypeSymbol.IsTypeParameterDisallowingAnnotation())
+                    {
+                        return NullableAnnotation.NotAnnotated;
+                    }
+                    return NullableAnnotation.Annotated;
+
+                case NullableAnnotation.NotNullable:
+                    // Example of unspeakable types:
+                    // - a "tight T", which is an unconstrained T which was null-tested already
+                    // - a "tight int?"
```

Comment:
Minor point: Consider avoiding introducing a new term. Perhaps just "// An unconstrained type parameter or nullable value type that is known to be not null".